### DISCUSSION TITLE
GraphQLSchema.description How can I access this in introspection?

### DISCUSSION THREAD

**Discussion by JimmyyyW** [graphql-java/graphql-java#2451] [Q&A] [answered]
Hi,

Just for some context, I'm actually working with graphql-kickstart, which of course consumes this library. I see the GraphQLSchema type has a description field and that top level schema descriptions have been added as a feature in various other GraphQL libraries (i.e Apollo for JS). 

When I go to introspect i.e:

```
query Introspect {
      __schema {
                //description??
                  type {
                      description
                  }
          }
}
```

Ideally I'd expect that property to be available there. How do I access this in an introspection query?

**UPDATE**
So it turns out it works exactly as intended with the above query (uncommenting description..), so I guess the question is actually if anyone knows how to configure typical clients i.e GraphiQL or equivalent to display this specifically in the markdown - eg (highlighted red):
<img width="1116" alt="Screenshot 2021-07-12 at 17 15 57" src="https://user-images.githubusercontent.com/50302042/125323760-10d47e80-e337-11eb-8640-fcdf6fb2dd30.png">

**Comment by bbakerman** [comment, marked as answer]
The  Introspection allowable fields are defined by the spec

https://spec.graphql.org/June2018/#sec-Schema-Introspection
```
type __Schema {
  types: [__Type!]!
  queryType: __Type!
  mutationType: __Type
  subscriptionType: __Type
  directives: [__Directive!]!
}
```

And it does not define a `description` on that type.  Your have stumbled onto something however.  It probably should.

But it doesn't and hence the error you see.  Its not an allow field on the introspection `__Schema` type

**Reply by JimmyyyW** [reply]
that's even more interesting, it's not part of the spec, does work in this library (the image I posted returns null because I didn't actually populate it, but if you look at `GraphQLSchema` class there is indeed a description property) however with further investigation I've found that the description field does exist on the schema introspection for `__schema` type in GraphiQL, but not insomnia (used above). The GraphiQL source code also contains `schema.description || A GraphQL schema provides a root type for each kind of operation` for populating that location, so I'm not 100% sure why that's not working.

So I guess the next step for me is to see if I can put the value of this property it where I highlighted either in GraphiQL or insomnia etc - may consider making that a contribution or raising that as it seems it's more so on the UI tools, regardless as to whether or not it's a requirement in the spec (which IMO it would make a lot of sense for it to be added there given documentation being a 1st class citizen)

**Reply by bbakerman** [reply]
Actually looking into this - we DO indeed have a GraphqlSchema.getDescription() and a field in `__Schema` for it

This is a graphql-java thing in anticipation of a spec change - which is in th working draft

https://spec.graphql.org/draft/#sec-The-__Schema-Type

however some tools like graphIQl HAVE hard coded their __Schema types because...well... they need to  - how can you ask a graphql system what types there are if you don't know the `__XXX` types that lead to type discovery

**Reply by JimmyyyW** [reply]
Ah I wasn't aware this was in draft. Thanks for this clarification, it's greatly appreciated! 

I'm mitigating the issue temporarily by enhancing introspection query from GraphiQL to contain description, in which case I'm getting the desired behaviour. Given it's in draft, I'll likely let that suffice as no doubt this will be an 'out-of-the-box' feature synergising across various products in the future
